modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/CallFunction.java

```diff
@@ -189,13 +189,15 @@ protected void setDDMFormFieldValues(
 
 		for (Map.Entry<String, String> entry : resultMap.entrySet()) {
 			String outputName = entry.getValue();
+			String ddmFormFieldName = entry.getKey();
 
 			if (!ddmDataProviderResponse.hasOutput(outputName)) {
+				setDDMFormFieldOptions(
```

Comment:
Ideally, we should be certain that the field uses options. Take a numeric field being used as data provider output as an example.